session.go

```diff
@@ -355,15 +356,15 @@ func (s *session) run() error {
 
 runLoop:
 	for {
+		s.maybeResetTimer()
```

Comment:
Please keep this below the error abort condition. We're potentially in an invalid state if we have an error, and we should do as little work as possible in that case.

Reply:
Fixed.